src/Carousel.js

```diff
@@ -280,7 +280,7 @@ const Carousel = React.createClass({
       direction = direction || this.getDirection(previousActiveIndex, index);
 
       if (this.props.onSelect) {
-        this.props.onSelect(index, direction);
+        this.props.onSelect(index, e, direction);
```

Comment:
This is only compatible with the rest rather than absolutely consistent.

I feel like there ought to be some value in preserving the `direction` argument, though.

Then again, I don't use `<Carousel>`.